transports/dns/src/lib.rs

```diff
@@ -110,85 +120,28 @@ const MAX_DNS_LOOKUPS: usize = 32;
 /// result of a single `/dnsaddr` lookup.
 const MAX_TXT_RECORDS: usize = 16;
 
-/// A `Transport` wrapper for performing DNS lookups when dialing `Multiaddr`esses
-/// using `async-std` for all async I/O.
-#[cfg(feature = "async-std")]
-pub type DnsConfig<T> = GenDnsConfig<T, AsyncStdResolver>;
-
-/// A `Transport` wrapper for performing DNS lookups when dialing `Multiaddr`esses
-/// using `tokio` for all async I/O.
-#[cfg(feature = "tokio")]
-pub type TokioDnsConfig<T> = GenDnsConfig<T, TokioAsyncResolver>;
-
 /// A `Transport` wrapper for performing DNS lookups when dialing `Multiaddr`esses.
+/// You shouldn't need to use this type directly. Use [`tokio::Config`] or [`async_std::Config`] instead.
 #[derive(Debug)]
-pub struct GenDnsConfig<T, R> {
+pub struct Config<T, R> {
```

Comment:
This should be called `Transport` too:

```suggestion
pub struct Transport<T, R> {
```